x/marker/keeper/send_restrictions.go

```diff
@@ -146,3 +146,44 @@ func ContainsWildCard(attr string) bool {
 	segs := strings.Split(attr, ".")
 	return len(segs) > 1 && segs[0] == "*"
 }
+
+// AddToRequiredAttributes add new attributes to current list, errors if attribute already exists
+func AddToRequiredAttributes(currentAttrs []string, addAttrs []string) ([]string, error) {
+	for _, ra := range addAttrs {
+		found := false
+		for _, add := range currentAttrs {
+			if ra == add {
+				found = true
+				break
+			}
+		}
+		if found {
+			return nil, fmt.Errorf("cannot add duplicate entry to required attributes %s", ra)
+		}
+		currentAttrs = append(currentAttrs, ra)
+	}
+	return currentAttrs, nil
+}
+
+// RemovesFromRequiredAttributes remove  attributes from current list, errors if attribute does not exists
+func RemovesFromRequiredAttributes(currentAttrs []string, removeAttrs []string) ([]string, error) {
+	expectedLen := len(currentAttrs) - len(removeAttrs)
+	reqAttrs := []string{}
+	for _, ra := range currentAttrs {
+		found := false
+		for _, cra := range removeAttrs {
+			if cra == ra {
+				found = true
+				break
+			}
+		}
+		if !found {
+			reqAttrs = append(reqAttrs, ra)
+		}
+	}
+	// check to see if there was an entry in remove list that did not exist
+	if len(reqAttrs) != expectedLen || expectedLen < 0 {
+		return removeAttrs, fmt.Errorf("remove required attributes list had incorrect entries")
+	}
+	return reqAttrs, nil
```

Comment:
Probably best to identify the entry in `removeAttrs` that isn't in `currentAttrs` for the error message.
```suggestion
	toRem := make(map[string]bool)
	for _, ra := range removeAttrs {
		found := false
		for _, ca := range currentAttrs {
			if ra == ca {
				found = true
				break
			}
		}
		if !found {
			return nil, fmt.Errorf("attribute %s is already not required", ra)
		}
		toRem[ra] = true
	}
	reqAttrs := make([]string, 0, len(currentAttrs))
	for _, ca := range currentAttrs {
		if !toRem[ca] {
			reqAttrs = append(reqAttrs, ca)
		}
	}
	return reqAttrs, nil
```